### ISSUE TITLE
Parallelize across cluster nodes for WGS

### ISSUE THREAD

**Issue by marchoeppner** [freebayes/freebayes#500]
Hi,

I'd like to use Freebayes for calling variants across 200 genomes (30X PE). I understand there are several things I can do to parallelize this, but am unsure about possible restrictions here. 

I was planning on using intervals for calling, for example those distributed by the Broad Institute for their production pipeline (several hundred intervals of a few kb up to dozens of MB). I would partition these (per compute node), and pass them as chr:start-stop via STDIN to freebayes-parallel (custom script, I suppose). 

Questions:

1) Is there a number of snps or interval size I should have within a given process (a job on a node via freebayes-parallel) to make any of the internal processing work better? I.e. would Freebayes work for 200 genomes running in parallel of 1 interval of 1MB. Or does it require larger genomic regions for calling? At the very least, I should probably have several intervals per job, so that freebayes-parallel actually works on multiple CPUs, correct?

2) Can i split contiguous intervals into smaller pieces for parallel processing, and if so, how does Freebayes deal with variants on the break points of these intervals? My concern is that Freebayes just takes a fairly long time to run on a 50MB interval across 200 genomes. So being able to split this into smaller bits would probably utilize our infrastructure a bit more efficiently. 

Any pointers would be greatly appreciated. 

Cheers,
Marc

**Comment by tseemann** [comment]
_"how does Freebayes deal with variants on the break points of these intervals"_

I think @ekg has said previously that all reads spanning the breakpoint take part in the call, so there is no problem with weird results near breakpoints.

**Comment by ekg** [comment]
Well, this is my impression based on how I wrote it and what I think it
should do. It could be that there are problems, as I admittedly didn't
specifically address this. You can overlap the intervals by a read length
to ensure that calls are fully made twice, then restrict back to the
intervals for merging. This should all be coded up properly...
